### PR TITLE
Attempt to resolve issues with roles differences between k8 and oscp

### LINKED ISSUE #20: Kubernetes translation of roles.yaml
The roles in https://github.com/openshift/open-service-broker-sdk/blob/master/test-scripts/resources/roles.yaml do not currently work in a standalone Kubernetes cluster, this is because the it uses OpenShift specific constructs. In order to allow this SDK to work for both Kubernetes and OpenShift, these roles need to be translated into the ```rbac``` API https://kubernetes.io/docs/admin/authorization/rbac/

### PR DESCRIPTION
**Pull request by maleck13**
@bparees PTAL when you have a min. Should resolve https://github.com/openshift/open-service-broker-sdk/issues/20 I believe. Most of the changes are moving the files to the new dir structure. I tested the changes against my own kube cluster, but would be good to have someone else try it if possible as this is the first time I have done anything with the rbac stuff in kubernetes.

### CODE REVIEW

**Comment by bparees** [comment]
Can you remove the test-scripts/resources dir/files?

File: install/rc.yaml [hunk 1]
```diff
@@ -22,8 +22,8 @@ spec:
         emptyDir: {}
       containers:
       - name: brokersdk
-        image: brokersdk
-        imagePullPolicy: Never
+        image: maleck13/service-broker-mongo:1.4
```
Comment (bparees) [inline install/rc.yaml]: looks like you caught a personal change here.

i'm fine w/ the change to IfNotPresent on the pull policy though (but i believe that's the default, so you can just remove it entirely)

Comment (maleck13) [reply install/rc.yaml]: yep 👍 That was me testing it :)

File: install/rc.yaml [hunk 2]
```diff
@@ -23,7 +23,6 @@ spec:
       containers:
       - name: brokersdk
         image: brokersdk
-        imagePullPolicy: Never
```
Comment (bparees) [inline install/rc.yaml:26]: i'm afraid this appears to be defaulting to a pull policy of "always" now.  sorry.  I could swear k8s only did that if the tag was explicitly ":latest".  Anyway please set it to IfNotPresent

**Comment by enj** [comment]
This is the wrong approach.  It is nearly impossible to manually convert to RBAC and guarantee that everything is in sync.  Write a small helper program to handle the conversion as that will save you time and work in the future.

**Comment by maleck13** [comment]
@enj Thanks for response. Do you have an example in a code base that I could use for inspiration or is this something that would need to be done from scratch. 
@bparees Does this PR provide enough value and we can create a separate issue for the tool, or would you prefer we keep this one open?

**Comment by maleck13** [comment]
@enj sorry missed your link to the code in the related issue. I will take a look at what you linked to

**Comment by maleck13** [comment]
@bparees @enj Hey guys starting to put a tool together.  Little unsure about decoding the objects. I have used the ```UniversalDecoder``` and ``` UnstructuredJSONScheme``` is there a way to decode objects like the ClusterRole into their correct types currently only getting *runtime.Unstructured.
gist of my **very** rough code to show how I am decoding https://gist.github.com/maleck13/fac92043989a8a425c247e5e6f4f0432#file-convert-go-L75

**Comment by enj** [comment]
You are missing the [imports](https://github.com/openshift/origin/blob/master/tools/etcdhelper/etcdhelper.go#L19-L22) that register objects.  That file has logic for decoding objects as well.

**Comment by maleck13** [comment]
@enj excellent thanks

**Comment by enj** [comment]
As a side note, I feel like the correct approach would be to define (in Go, not YAML) the RBAC roles you need.  Then convert them to Openshift roles (in Go code).  Then dump both of list as YAML.  So the input to the tool is Go code, not YAML.

**Comment by pmorie** [comment]
In general I think we should install the broker with a technology conventional for the environment it's being installed into.  For k8s, many people are going to want a helm chart to install their broker.  On OpenShift, people are going to expect templates.  I think there's probably enough divergence in the roles to maintain both an OpenShift template and a k8s template.

I am not saying no to this PR, but stating my own opinion about what this dimension of the experience should be like for users.

**Comment by maleck13** [comment]
@pmorie @enj @bparees How about we take this PR as an incremental improvement? Which it is as the roles and installation didn't work on k8s at all and now it does. We could then add new issues for helm installation and the roles generation tool?

**Comment by bparees** [comment]
> @pmorie @enj @bparees How about we take this PR as an incremental improvement? Which it is as the roles and installation didn't work on k8s at all and now it does. We could then add new issues for helm installation and the roles generation tool?

i'm perfectly happy with that approach.

**Comment by enj** [comment]
SGTM

File: install/openshift/roles.yaml [hunk 3]
```diff
@@ -2,6 +2,23 @@ apiVersion: v1
 kind: List
 items:
 - apiVersion: v1
+  kind: ClusterRole
```
Comment (bparees) [inline install/openshift/roles.yaml]: this cluster role is a default part of openshift now so we don't need it anymore.

**Review by bparees — CHANGES_REQUESTED** [review]
remove the unnecessary clusterrole and i'll merge.

**Comment by maleck13** [comment]
@bparees made that change

**Comment by bparees** [comment]
awesome, thanks @maleck13!

**Pull request merged**